jul-ecs-formatter/src/main/java/co/elastic/logging/jul/FallbackMdc.java

```diff
@@ -0,0 +1,71 @@
+/*-
+ * #%L
+ * Java ECS logging
+ * %%
+ * Copyright (C) 2019 - 2022 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.logging.jul;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Fallback MDC implementation to be used when JUL is deployed without one
+ */
+public class FallbackMdc {
+
+    public static final FallbackMdc INSTANCE = new FallbackMdc();
+
+    private final InheritableThreadLocal<Map<String, String>> tlm = new InheritableThreadLocal<Map<String, String>>() {
```

Comment:
Why `InheritableThreadLocal` instead of `ThreadLocal`?

Reply:
I took inspiration from `org.slf4j.helpers.BasicMDCAdapter` (slf4j implementation for JUL) which also relies on it, also the logback adapter also does the same. adding a few tests to see when/if it's required might be a good option here.

Reply:
Inheritable thread local have been removed because usage by java agent does not require it, it's usually used within the scope of a single method call within logging framework/lib, hence thread creation and delegation is not used at all.